src/System.Private.DataContractSerialization/src/System/Xml/StringHandle.cs

```diff
@@ -227,15 +228,17 @@ private bool Equals2(int offset2, int length2, XmlBufferReader bufferReader2)
             return GetString() == _bufferReader.GetString(offset2, length2);
         }
 
-        private bool Equals2(StringHandle s2)
+        public bool Equals(StringHandle other)
         {
-            StringHandleType type = s2._type;
+            if (ReferenceEquals(other, null))
```

Comment:
Possibly. It's possible that this doesn't come up with this class, which is after all internal and hence restricted in the range of values it could see. The tests that exercise this class (and PrefixHandle) don't hit that case, but it could be that they should do and that points to a gap in the tests, but since they don't I'd rather keep this PR having only changes that can be shown as being a win by very local examination only.